app/src/main/java/com/sdp13epfl2021/projmag/activities/ProfilePageActivity.kt

```diff
@@ -36,7 +36,7 @@ class ProfilePageActivity : AppCompatActivity() {
             //buttonAddCv.setVisibility(View.INVISIBLE)
         }
 
-        UserProfileDatabase(Firebase.firestore, Firebase.auth).getProfile(::loadUserProfile) {
+        FirebaseUserdataDatabase(Firebase.firestore, Firebase.auth).getProfile(::loadUserProfile) {
```

Comment:
Instead of creating a new FirebaseUserdataDatabase at each call, we can use Utils.getInstance(this).userdataDatabase (until depency injection is made), and store the database.

Reply:
Utils.getInstance(this).userdataDatabase uses cachedDatabase methods which have not been implemented yet for getting and uploading profiles. If in both cases the code will change when we implement dependency injection, I will keep it like it was